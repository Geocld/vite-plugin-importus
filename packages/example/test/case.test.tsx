import React from 'react'
import renderer from 'react-test-renderer'
import Case1 from '../src/components/Case1'
import Case2 from '../src/components/Case2'
import Case3 from '../src/components/Case3'
import Case4 from '../src/components/Case4'
import Case5 from '../src/components/Case5'
import Case6 from '../src/components/Case6'

function toJson(component) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result
}

test('Test Case 1', () => {
  const component = renderer.create(
    <Case1/>,
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Test Case 2', () => {
  const component = renderer.create(
    <Case2/>,
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Test Case 3', () => {
  const component = renderer.create(
    <Case3/>,
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Test Case 4', () => {
  const component = renderer.create(
    <Case4/>,
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Test Case 5', () => {
  const component = renderer.create(
    <Case5/>,
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Test Case 6', () => {
  const component = renderer.create(
    <Case6/>,
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})