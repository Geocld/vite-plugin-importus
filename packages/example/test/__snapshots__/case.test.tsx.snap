// Vitest Snapshot v1

exports[`Test Case 1 1`] = `
<div
  className="container"
>
  <div
    className="btns"
  >
    <button
      className="ant-btn ant-btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        Primary
      </span>
    </button>
  </div>
  <div
    className="tags"
  >
    <span
      className="ant-tag"
      style={
        {
          "backgroundColor": undefined,
        }
      }
    >
      Tag1
    </span>
    <span
      className="ant-tag"
      selected={true}
      style={
        {
          "backgroundColor": undefined,
        }
      }
    >
      Tag2
    </span>
  </div>
</div>
`;

exports[`Test Case 2 1`] = `
<div
  className="container"
>
  <div
    className="btns"
  >
    <button
      className="ant-btn ant-btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        Primary
      </span>
    </button>
  </div>
  <div
    className="tags"
  >
    <span
      className="ant-tag"
      style={
        {
          "backgroundColor": undefined,
        }
      }
    >
      Tag1
    </span>
    <span
      className="ant-tag"
      selected={true}
      style={
        {
          "backgroundColor": undefined,
        }
      }
    >
      Tag2
    </span>
  </div>
</div>
`;

exports[`Test Case 3 1`] = `
<div
  className="container"
>
  <div
    className="btns"
  >
    <button
      className="ant-btn ant-btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        Primary
      </span>
    </button>
  </div>
  <div
    className="tags"
  >
    <span
      className="ant-tag"
      style={
        {
          "backgroundColor": undefined,
        }
      }
    >
      Tag1
    </span>
    <span
      className="ant-tag"
      selected={true}
      style={
        {
          "backgroundColor": undefined,
        }
      }
    >
      Tag2
    </span>
  </div>
</div>
`;

exports[`Test Case 4 1`] = `
<div
  className="container"
>
  <div
    className="btns"
  >
    <button
      className="ant-btn ant-btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        Primary
      </span>
    </button>
  </div>
  <div
    className="tags"
  >
    <span
      className="ant-tag"
      style={
        {
          "backgroundColor": undefined,
        }
      }
    >
      Tag1
    </span>
    <span
      className="ant-tag"
      selected={true}
      style={
        {
          "backgroundColor": undefined,
        }
      }
    >
      Tag2
    </span>
  </div>
</div>
`;

exports[`Test Case 5 1`] = `
<div
  className="container"
>
  <div
    className="btns"
  >
    <button
      className="ant-btn ant-btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        Primary
      </span>
    </button>
  </div>
  <div
    className="tags"
  >
    <span
      className="ant-tag"
      style={
        {
          "backgroundColor": undefined,
        }
      }
    >
      Tag1
    </span>
    <span
      className="ant-tag"
      selected={true}
      style={
        {
          "backgroundColor": undefined,
        }
      }
    >
      Tag2
    </span>
  </div>
</div>
`;

exports[`Test Case 6 1`] = `
<div
  className="container"
>
  <div
    className="btns"
  >
    <button
      className="ant-btn ant-btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        Primary
      </span>
    </button>
  </div>
  <div
    className="tags"
  >
    <span
      className="ant-tag"
      style={
        {
          "backgroundColor": undefined,
        }
      }
    >
      Tag1
    </span>
    <span
      className="ant-tag"
      selected={true}
      style={
        {
          "backgroundColor": undefined,
        }
      }
    >
      Tag2
    </span>
  </div>
</div>
`;
